components/raftstore/src/store/fsm/apply.rs

```diff
@@ -522,9 +543,17 @@ where
             self.perf_context.start_observe();
             let mut write_opts = engine_traits::WriteOptions::new();
             write_opts.set_sync(need_sync);
-            self.kv_wb_mut().write_opt(&write_opts).unwrap_or_else(|e| {
+            write_opts.set_disable_wal(self.disable_wal);
```

Comment:
Note, if WAL is disabled, then it needs to be disabled in all places. Otherwise, after restart, it may cause corruption.